### PR TITLE
data to server adapted to changed API

### PR DESCRIPTION
**Pull request by rexfue**
see #189

### CODE REVIEW

**Comment by rexfue** [comment]
OK prima - failed. Und was muss such da jetzt machen? Wie akzeptiere ich, was der geändert haben will?

**Comment by ThomasWaldmann** [comment]
Local `astyle ...` (siehe unsere dev docs) laufen lassen und die Aenderungen committen / pushen.

**Comment by rexfue** [comment]
OK, done.
Aber was bringt dann ganze travis, wenn ich eh alles selber von Hand machen muss?

**Comment by ThomasWaldmann** [comment]
Es stellt sicher, dass die Tests laufen und findet dabei Probleme, die sonst oefters mal nicht bemerkt werden. In unserem Fall aktuell eben "Style-Probleme" und "kompiliert nicht"-Probleme.

Man kann das auch erweitern und wirklich die korrekte Funktion von Code-Teilen testen (in unserem Fall nur einfach bei den Teilen, die keine HW brauchen).

File: multigeiger/multigeiger.ino [hunk 1]
```diff
@@ -78,6 +78,11 @@
 #define CRITICAL 4
 #define NOLOG 999  // only to set log_level, so log() never creates output
 
+// Sensor-PINS. These definitions ar for sensor.community. They decide the kind of the sensor using this number.
```
[the remark this answers is not in the document]
Comment (rexfue) [reply multigeiger/multigeiger.ino]: Yes, that's their term. I would keep it. I`ll explain in more detail.

File: multigeiger/multigeiger.ino [hunk 2]
```diff
@@ -894,63 +897,51 @@ void jbTone(unsigned int frequency_mHz, unsigned int time_ms, unsigned char volu
...
-  body += ",{\"value_type\":\"sample_time_ms\",\"value\":\"" + String(timediff) + "\"}";
-  body += "]}";
-  if (DEBUG_SERVER_SEND == 1) {
-    log(DEBUG, "body: %s", body.c_str());
-  }
-  return body;
+static void add_Value2Json(String &res, const String &type, const String &value) {
+  String s = F("{\"value_type\":\"{t}\",\"value\":\"{v}\"},");
+  s.replace("{t}", type);
+  s.replace("{v}", value);
```
[the remark this answers is not in the document]
Comment (rexfue) [reply multigeiger/multigeiger.ino]: Yeah, right. It was a copy/paste from airrohr-firmware

File: multigeiger/multigeiger.ino [hunk 3]
```diff
@@ -894,63 +897,51 @@ void jbTone(unsigned int frequency_mHz, unsigned int time_ms, unsigned char volu
...
+    add_Value2Json(body, counts, String(GMC_counts_2send));
+    add_Value2Json(body, sample_time_ms, String(timediff));
+  }
+  if ((sendwhat == SEND_BME) || (sendwhat == SEND_ALL)) {
+    add_Value2Json(body, String((addname ? "BME280_" : "")) + "temperature", String(bme_temperature, 2));
+    add_Value2Json(body, String((addname ? "BME280_" : "")) + "humidity", String(bme_humidity, 2));
+    add_Value2Json(body, String((addname ? "BME280_" : "")) + "pressure", String(bme_pressure, 2));
+  }
+  body.remove(body.length() - 1);
+  body += "]}";
```
[the remark this answers is not in the document]
Comment (rexfue) [reply multigeiger/multigeiger.ino]: don't understand. example?

File: multigeiger/multigeiger.ino [hunk 4]
```diff
@@ -654,29 +659,27 @@ void loop() {
     #if SEND2DUMMY
     displayStatusLine("Toilet");
     log(INFO, "SENDING TO TOILET ...");
-    sendData2http(TOILET, SEND_CPM, hvp, time_difference, true);
+    sendData2http(TOILET, String(ssid), SEND_ALL, hvp, time_difference);
```
[the remark this answers is not in the document]
Comment (rexfue) [reply multigeiger/multigeiger.ino]: sensor.community/ madam distinguishes the sensors by this number in the http header (see readme)

[the remark this answers is not in the document]
Comment (ThomasWaldmann) [reply multigeiger/multigeiger.ino]: sorry, confused the AP SSID (this) with the client SSID (which should not be transmitted).

File: multigeiger/multigeiger.ino (hunk 3 above)
[the remark this answers is not in the document]
Comment (ThomasWaldmann) [reply multigeiger/multigeiger.ino]: like OPENING + CONTENT + CLOSING.

[the remark this answers is not in the document]
Comment (ThomasWaldmann) [reply multigeiger/multigeiger.ino]: if we do more json stuff, some json lib would be a better option than handmade json.

File: multigeiger/multigeiger.ino (hunk 1 above)
[the remark this answers is not in the document]
Comment (ThomasWaldmann) [reply multigeiger/multigeiger.ino]: ok. guess it is a PART IDENTIFICATION NUMBER or so.

File: multigeiger/multigeiger.ino (hunk 3 above)
[the remark this answers is not in the document]
Comment (rexfue) [reply multigeiger/multigeiger.ino]: Good idea, I'll check/try Arduino JSON

File: multigeiger/multigeiger.ino (hunk 1 above)
[the remark this answers is not in the document]
Comment (rexfue) [reply multigeiger/multigeiger.ino]: yes

**Comment by ThomasWaldmann** [comment]
If there is a problem fixing the conflicts here, I can help.

Just push ALL the changes you have for this to github and tell me.

**Comment by ThomasWaldmann** [comment]
If adding the json lib delays this PR a lot, we could also do that later and open a ticket for it.

I'ld like to get this PR merged ASAP, so I can also finish #200.

**Comment by rexfue** [comment]
OK, move on. 
I'm stuck with other problems and will work on that later this week.

**Comment by rexfue** [comment]
closing this PR.
generate a new on built on new master

**Pull request closed without merging**